parastell/radial_distance_utils.py

```diff
@@ -1,154 +1,360 @@
+                np.concatenate(
+                    [coil.coords[:-1:sample_mod], [coil.coords[0]]]
+                ),
+                np.concatenate(
+                    [next_coil.coords[:-1:sample_mod], [next_coil.coords[0]]]
+                ),
             )
+        ]
+        for coil, next_coil in zip(magnet_coils[:-1], magnet_coils[1:])
+    ]
+
+    surface_lines = np.array(surface_lines)
+    surface_sections = np.reshape(
+        surface_lines,
+        (
+            len(magnet_coils) - 1,
+            len(magnet_coils[0].coords[:-1:sample_mod]) + 1,
+        ),
+    )
 
-        filaments[filament_index] = reordered_filament
+    [
+        [
+            cubit.cmd(f"create surface skin curve {line} {next_line}")
+            for line, next_line in zip(lines[:-1], lines[1:])
+        ]
+        for lines in surface_sections
+    ]
 
-    filaments = sort_filaments_toroidally(filaments)
 
-    return filaments
+def fire_ray(point, direction):
+    """Performs a ray-firing operation in Coreform Cubit from a reference point
+    and along a reference direction to determine the distance from that point
+    to the magnet surface.
 
+    Arguments:
+        point (iterable of float): Cartesian coordinates of reference point.
+        direction (iterable of float): reference direction in
+            Cartesian-coordinate system.
 
-def get_reordered_filaments(magnet_set):
+    Returns:
+        distance (float): distance between reference point and magnet surface,
+            along reference direction.
     """
-    Convenience function to get the reordered filament data from a magnet
+    cubit.cmd(f"create vertex {point[0]} {point[1]} {point[2]}")
+    vertex_id = cubit.get_last_id("vertex")
+
+    cubit.cmd(
+        f"create curve location at vertex {vertex_id} "
+        f"location fire ray location at vertex {vertex_id} "
+        f"direction {direction[0]} {direction[1]} {direction[2]} at "
+        "surface all maximum hits 1"
+    )
+
+    curve_id = cubit.get_last_id("curve")
+    distance = cubit.get_curve_length(curve_id)
+
+    return distance
+
+
+def measure_surface_coils_separation(surface):
+    """Determines the distance between a given Surface class object and a
+    surface spanning a set of MagnetCoil class objects via a ray-firing
+    operation in Coreform Cubit.
+
+    Arguments:
+        surface (object): Surface class object.
+
+    Returns:
+        distance_matrix (2-D np.array of float): matrix of distances between
+            points defining surface.Ribs and magnet surface, along
+            Rib._normals.
     """
-    magnet_set.populate_magnet_coils()
+    distance_matrix = np.array(
+        [
+            [
+                fire_ray(point, direction)
+                for point, direction in zip(rib.rib_loci, rib._normals())
+            ]
+            for rib in surface.Ribs
+        ]
+    )
+
+    return distance_matrix
+
+
+def measure_fw_coils_separation(
```

Comment:
If the first layer in the radial build is called "sol" does it get merged into chamber if split_chamber is false?

Reply:
No, the layer would have to be named “chamber”